### ISSUE TITLE
Inconsistent boolean parameter in API `/source` endpoint

### ISSUE THREAD

**Issue by StayPirate** [openSUSE/open-build-service#10789]
# Issue Description
The API endpoint `/source` treat boolean parameters `deleted` as a flag. If it's present within the querystring, then it is evaluated **True** regardless of the assigned value: `?deleted`, `?deleted=1`, `?deleted=0` are all **True**.
This is inconsistent with other endpoints, where boolean parameters are evaluated.

Expected Result
============
I would expect only the second HTTP request returns a **403 Forbidden**, while I expect to get a **200 OK** from the first.

```bash
$ curl -su $OBS_AUTH -G "https://api.suse.de/source?deleted=0"
<status code="no_permission_for_deleted">
  <summary>only admins can see deleted projects</summary>
</status>

$ curl -su $OBS_AUTH -G "https://api.suse.de/source?deleted=1"
<status code="no_permission_for_deleted">
  <summary>only admins can see deleted projects</summary>
</status>
```

**Comment by StayPirate** [comment]
Adding @crazyscientist who's interested in it and addressed me to report this here.

**Comment by eduardoj** [comment]
Hi @StayPirate, thanks for reporting the issue.

In the API endpoint `/source`, the `deleted` parameter is only available for Admins.

From the errors you are receiving (you always receive `<status code="no_permission_for_deleted">`), it looks like you are querying this API endpoint without being an admin. This is already documented in the the API documentation. Search for the `deleted` parameter in this endpoint: https://api.opensuse.org/apidocs/#/Sources%20-%20Projects/get_source

Just to make things more clear, making use of the `deleted` parameter results in checking for credentials of being Admin. And that means, a `no_permission_for_deleted` with a 403 status will be returned for a non-admin user for all these endpoints:
- https://api.suse.de/source?deleted
- https://api.suse.de/source?deleted=0
- https://api.suse.de/source?deleted=1

By other hand, the inconsistency boolean parameter issue can be included into the recently created issues #14651 and #14652.

I'm closing this one in favor of the recent ones, which are more generic.
